### ISSUE TITLE
It's not possible to track requests made with excon using string as HTTP Method

### ISSUE THREAD

**Issue by nickrisaro** [open-telemetry/opentelemetry-ruby#1158]
**Description of the bug**
The excon instrumentation shows an error if an string is used as value for the method. According to the [excon docs](https://www.rubydoc.info/gems/excon/0.79.0#:~:text=request%20takes%20a%20method%20option%2C%20accepting%20either%20a%20symbol%20or%20string) this is allowed.

The error on the logs is
`[2022-03-22T17:31:14.333447 #85108] ERROR -- : OpenTelemetry error: invalid span attribute value type NilClass for key 'http.method' on span 'HTTP '`

The code only checks for HTTP Methods set as symbols [here](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/instrumentation/excon/lib/opentelemetry/instrumentation/excon/middlewares#L28)

**Share details about your runtime**

Operating system details: MacOS, Monterey 12.2.1
RUBY_ENGINE: "ruby"
RUBY_VERSION: "2.7.5"
RUBY_DESCRIPTION: "ruby 2.7.5p203 (2021-11-24 revision f69aeb8314) [x86_64-darwin20]"

**Share a simplified reproduction if possible**

```rb
require 'bundler/inline'

gemfile(true) do
  source 'https://rubygems.org'

  gem 'opentelemetry-api'
  gem 'opentelemetry-sdk'
  gem 'opentelemetry-instrumentation-all'
  gem 'excon'
end

require 'opentelemetry-api'
require 'opentelemetry-sdk'
require 'excon'

OpenTelemetry::SDK.configure do |c|
  c.use_all
end

Excon.new('http://google.com').request(:path => '/', :method => 'GET')
```

**Comment by plantfansam** [comment]
I mentioned this in #1159, but I believe this should be addressed by https://github.com/open-telemetry/opentelemetry-ruby/pull/1147/

**Comment by fbogsany** [comment]
Fixed in #1147
